mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -358,6 +360,25 @@ pub(crate) async fn block_header_utxo_loop<T: UtxoCommonOps>(
                 );
                 // Todo: remove this electrum server and use another in this case since the headers from this server can't be retrieved
                 sync_status_loop_handle.notify_on_permanent_error(error);
+                Timer::sleep(args.error_sleep).await;
+                break;
+            },
+        };
+
+        // Detect and resolve header for chain reorg
+        match detect_and_resolve_chain_reorg(ticker, from_block_height, storage.inner.as_ref(), &block_headers).await {
```

Comment:
Before checking if there was a reorganization, the new headers should be validated (checked that they achieve the required difficulty) to make sure they are real. In general, the chain with the highest difficulty is the best chain, so a comparison with the conflicting headers difficulties is required too.

Reply:
ok